NDTensors/src/blocksparse/blocksparse.jl

```diff
@@ -38,10 +21,17 @@ function BlockSparse(
   )
 end
 
+function BlockSparse(
+  datatype::Type{<:UnallocatedZeros}, blockoffsets::BlockOffsets, dim::Integer; vargs...
+)
+  return BlockSparse(datatype((dim,)), blockoffsets; vargs...)
+end
```

Comment:
I guess we need this since the version above that uses `fill!` won't work for `UnallocatedZeros`? That's a bit unfortunate. I wonder if something like this would work generically:
```julia
function BlockSparse(
  datatype::Type{<:AbstractArray}, blockoffsets::BlockOffsets, dim::Integer; vargs...
)
  return BlockSparse(
    generic_zeros(datatype, dim), blockoffsets; vargs...
  )
end
```
and then we would just need to make sure `generic_zeros` works properly for `UnallocatedZeros`.

Reply:
Also makes me think that we should rename `generic_zeros` to `similar_zeros` but that could be done later.

Reply:
Yeah I have a comment in that code because it is confusing. But yes I can use generic_zeros. My thought here was that in the next PR I will be reworking the blocksparse code and will have to make changes there so I was just getting things to work for now. But I don't see any issue using generic_zeros

Reply:
I have the generic_zeros code hacked together to work. We still need to be able to specify_parameters for the UnallocatedZeros type, but I have started working on that